### DISCUSSION TITLE
How to setup unique constraint with soft delete enabled

### DISCUSSION THREAD

**Discussion by talgat-ruby** [hasura/graphql-engine#7799] [Question] [answered]
Basically with Postgres `NULL != NULL`. And adding
```
( name, surname, deleted_at ) - person_name_surname_deleted_at_key
```
won't work.

In raw query uniqueness can be achieved with
```
(name, surname, COALESCE(deleted_at, ''))
```

Is there workaround for Hasura?

**Comment by jdinartejesus** [comment]
@talgat-ruby did you found a way?

**Reply by talgat-ruby** [reply, marked as answer]
Hey @jdinartejesus,

If you are using postgres 15 or newer you can use new constraint `UNIQUE NULLS NOT DISTINCT`. If not than it is possible only via workaround like not using null. 

More on [stackoverflow question](https://stackoverflow.com/questions/8289100/create-unique-constraint-with-null-columns).

**Reply by jdinartejesus** [reply]
Worked! Thanks so much for the quick reply!
